src/basic/virt.c

```diff
@@ -796,6 +814,13 @@ Virtualization detect_container(void) {
         if (r < 0)
                 log_debug_errno(r, "Failed to detect cgroup namespace: %m");
 
+        /* Finally, the root pid namespace has an hardcoded inode number of 0xEFFFFFFC since kernel 3.8, so
+         * if all else fails we can check the inode number of our pid namespace and compare it. */
+        if (running_in_pidns() > 0) {
+                v = VIRTUALIZATION_CONTAINER_OTHER;
```

Comment:
At least there are two possibilities we result container-other, when we detect cgroupns, or pidns. So, maybe better to add debugging logs in what sense we conclude the system as container-other.